### PR TITLE
Allow for other CE instance + create URL

### LINKED ISSUE #1: Allow different remote url other than godbolt.org
CE is selfhosted by some users. I think they would appreciate if there was a way to specify other remote url. Maybe even an IP to use a local instance! :)

### PR DESCRIPTION
**Pull request by dkm**
This should fix #1 and #3 .

I've also took the liberty of using a different mecanism for building JSON that is sent over the POST. It had the advantage to be easily copy/pasted across the python and shell script I'm already using :D

### CODE REVIEW

File: src/requests.rs
Comment (emmatyping) [inline src/requests.rs:33]: Wow, this is a lot nicer. Thanks!

File: src/url.rs
Comment (emmatyping) [inline src/url.rs]: Speaking of easy copy/pasting, I think you left this in and didn't mean to :)

**Review by emmatyping — APPROVED** [review]
This looks really nice. Thank you!

File: src/url.rs
Comment (dkm) [reply src/url.rs]: Yes, that's true. There is so much that could be fine tuned :) I can rebase and remove this line...

**Comment by dkm** [comment]
rebased and pushed !

**Comment by emmatyping** [comment]
Excellent, thank you! I will merge once CI go green.

**Comment by dkm** [comment]
I've added a single commit for updating the README (very basic).

**Comment by emmatyping** [comment]
Thank you so much!

**Comment by dkm** [comment]
Thanks for merging :)

For my information, how did you merge it ? Usually, merging PR maintains original commits (and authorship -- in this case, github changed the mail), but it seems that in this case all my commits were squashed before merging ? It's not really a problem in this case as it's rather small, but more by curiosity.

**Comment by emmatyping** [comment]
There is an option to squash and merge on Github if you hit the arrow next to the merge button. It's the default several other projects I work on use, so its somewhat become habit for me :)

FWIW authorship stays the same, it just mentions I committed.
image

**Comment by dkm** [comment]
Ok 😃 But authorship is changed. In my case, my workmail has been changed by my personnal mail. It's good to know as it can be an issue for attribution (ie. when work being contributed is paid by my employer, he wants to be credited).

**Comment by emmatyping** [comment]
Ah, well if you want I wouldn't be opposed to adding an ACKS file for acknowledgements.

**Comment by dkm** [comment]
No no, that's fine 😃

**Pull request merged**
